### ISSUE TITLE
Bug: update and list verbs are very slow to start

### ISSUE THREAD

**Issue by notevenaperson** [ArchiveBox/ArchiveBox#1066]
#### Describe the bug
On a large archive, `archivebox update` or `archivebox list` starts right away some CPU intensive process that takes a long time to complete. 
#### Steps to reproduce
1. Have a large archive (I have 1800+ links)
2. `archivebox update` or `list`
#### Screenshots or log output
```
docker-compose run archivebox update
[i] [2022-12-21 23:02:15] ArchiveBox v0.6.2: archivebox update
    > /data

[▶] [2022-12-21 23:05:07] Starting archiving of 1847 snapshots in index...
```

The user is simply left waiting, and for the lack of an explanation, the user is asked of his faith. Fans are screeching, it could be an infinite loop. If the user waits, they would wait 3 minutes. If they kill the program, the suspicion of an infinite loop would be wrong: (who could blame them)

```
Traceback (most recent call last):
  File "/usr/local/bin/archivebox", line 33, in <module>
    sys.exit(load_entry_point('archivebox', 'console_scripts', 'archivebox')())
  File "/app/archivebox/cli/__init__.py", line 140, in main
    run_subcommand(
  File "/app/archivebox/cli/__init__.py", line 80, in run_subcommand
    module.main(args=subcommand_args, stdin=stdin, pwd=pwd)    # type: ignore
  File "/app/archivebox/cli/archivebox_update.py", line 119, in main
    update(
  File "/app/archivebox/util.py", line 114, in typechecked_function
    return func(*args, **kwargs)
  File "/app/archivebox/main.py", line 752, in update
    matching_folders = list_folders(
  File "/app/archivebox/util.py", line 114, in typechecked_function
    return func(*args, **kwargs)
  File "/app/archivebox/main.py", line 892, in list_folders
    return STATUS_FUNCTIONS[status](links, out_dir=out_dir)
  File "/app/archivebox/index/__init__.py", line 407, in get_indexed_folders
    links = [snapshot.as_link_with_details() for snapshot in snapshots.iterator()]
  File "/app/archivebox/index/__init__.py", line 407, in <listcomp>
    links = [snapshot.as_link_with_details() for snapshot in snapshots.iterator()]
  File "/app/archivebox/core/models.py", line 127, in as_link_with_details
    return load_link_details(self.as_link())
  File "/app/archivebox/util.py", line 114, in typechecked_function
    return func(*args, **kwargs)
  File "/app/archivebox/index/__init__.py", line 347, in load_link_details
    return merge_links(existing_link, link)
  File "/app/archivebox/util.py", line 114, in typechecked_function
    return func(*args, **kwargs)
  File "/app/archivebox/index/__init__.py", line 107, in merge_links
    history[method] = list(reversed(sorted(
  File "/app/archivebox/index/__init__.py", line 108, in <genexpr>
    (ArchiveResult.from_json(pyjson.loads(result)) for result in deduped_jsons),
  File "/app/archivebox/index/schema.py", line 96, in from_json
    info['start_ts'] = parse_date(info['start_ts'])
  File "/app/archivebox/util.py", line 114, in typechecked_function
    return func(*args, **kwargs)
  File "/app/archivebox/util.py", line 157, in parse_date
    return dateparser(date, settings={'TIMEZONE': 'UTC'}).replace(tzinfo=timezone.utc)
  File "/usr/local/lib/python3.9/site-packages/dateparser/conf.py", line 89, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/dateparser/__init__.py", line 54, in parse
    data = parser.get_date_data(date_string, date_formats)
  File "/usr/local/lib/python3.9/site-packages/dateparser/date.py", line 420, in get_date_data
    for locale in self._get_applicable_locales(date_string):
  File "/usr/local/lib/python3.9/site-packages/dateparser/date.py", line 469, in _get_applicable_locales
    if self._is_applicable_locale(locale, s):
  File "/usr/local/lib/python3.9/site-packages/dateparser/date.py", line 473, in _is_applicable_locale
    return locale.is_applicable(
  File "/usr/local/lib/python3.9/site-packages/dateparser/languages/locale.py", line 70, in is_applicable
    date_string = self._simplify(date_string, settings=settings)
  File "/usr/local/lib/python3.9/site-packages/dateparser/languages/locale.py", line 372, in _simplify
    simplifications = self._get_simplifications(settings=settings)
  File "/usr/local/lib/python3.9/site-packages/dateparser/languages/locale.py", line 379, in _get_simplifications
    no_word_spacing = eval(self.info.get('no_word_spacing', 'False'))
  File "<string>", line 1, in <module>
KeyboardInterrupt
```

Long story short, I did some set up the dev enviroment and snooped around. Before telling the user anything, archivebox is iterating on every matching link, of 1847. But that's not what's slow! It's a particular function, `merge_links()` that, ran 1847 times, adds up to a lot of waiting. 

Merge_links is called by load_link_details, seemingly to combine the information from disk about the link we're currently processing, and prettify it too. So far so good, but why are we doing in this in bulk? For example, archivebox list is going to iterate on each link, to print it, so why not "merge" it when it unrolls. Do we really need a complete list of merged links before doing anything? Perhaps I'm not seeing the entire picture though...

**Comment by pirate** [comment]
The merge is done before the list in order to dedupe them, as sometimes there are duplicate snapshots from an older install or other archive getting merged in.

I'll likely improve this in the future, but it might require splitting out the import/dedupe step into an explicit user-run command in order to unblock the performance changes.
